packages/eas-cli/src/commands/channel/delete.ts

```
@@ -0,0 +1,152 @@
+                  id
+                  name
+                }
+              }
+            }
+          }
+        `,
+        {
+          appId,
+          name,
+        },
+        { additionalTypenames: ['UpdateChannel'] }
+      )
+      .toPromise()
+  );
+  return data;
+}
+
+async function deleteChannelOnAppAsync({
+  channelId,
+}: DeleteUpdateChannelMutationVariables): Promise<DeleteUpdateChannelResult> {
+  const data = await withErrorHandlingAsync(
+    graphqlClient
+      .mutation<DeleteUpdateChannelMutation, DeleteUpdateChannelMutationVariables>(
+        gql`
+          mutation DeleteUpdateChannel($channelId: ID!) {
+            updateChannel {
+              deleteUpdateChannel(channelId: $channelId) {
+                id
+              }
+            }
+          }
+        `,
+        {
+          channelId,
+        }
+      )
+      .toPromise()
+  );
+  return data.updateChannel.deleteUpdateChannel;
+}
+
+export default class ChannelDelete extends EasCommand {
+  static hidden = true;
+  static description = 'Delete a channel on the current project';
+
+  static args = [
+    {
+      name: 'name',
+      required: false,
+      description: 'Name of the channel to delete',
+    },
+  ];
+  static flags = {
+    json: flags.boolean({
+      description: `return JSON with the edited branch's ID and name.`,
+      default: false,
+    }),
+  };
+
+  async runAsync(): Promise<void> {
+    let {
+      args: { name },
+      flags: { json: jsonFlag },
+    } = this.parse(ChannelDelete);
+    if (jsonFlag) {
+      enableJsonOutput();
+    }
+
+    const projectDir = await findProjectRootAsync();
+    const { exp } = getConfig(projectDir, { skipSDKVersionRequirement: true });
+    const fullName = await getProjectFullNameAsync(exp);
+    const projectId = await getProjectIdAsync(exp);
+
+    if (!name) {
+      const validationMessage = 'Channel name may not be empty.';
+      if (jsonFlag) {
```

Hmmm, I'm not sure if we want to throw an error if someone passes the `--json` flag. I'd prefer to add another flag - `--non-interactive` - and throw only then.